spec/features/stash_engine/dataset_queuing_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+require 'fileutils'
+
+# rubocop:disable Metrics/BlockLength
+RSpec.feature 'DatasetQueuing', type: :feature do
+
+  HOLD_SUBMISSIONS_PATH = File.expand_path(File.join(Rails.root, '..', 'hold-submissions.txt')).freeze
+
+  # include MerrittHelper
+  include DatasetHelper
+  include Mocks::Datacite
+  # include Mocks::Repository
+  include Mocks::SubmissionJob
+  include Mocks::RSolr
+  include Mocks::Ror
+  include Mocks::Stripe
+  include AjaxHelper
+
+  before(:each) do
+    FileUtils.rm(HOLD_SUBMISSIONS_PATH) if File.exist?(HOLD_SUBMISSIONS_PATH)
+    # mock_repository!
+    # for this we don't want to mock the whole repository, but just the actual submission to Merritt that happens in
+    # the queue, Stash::Merritt::SubmissionJob.do_submit!
+    mock_submission_job!
+    mock_solr!
+    mock_ror!
+    mock_datacite!
+    mock_stripe!
+    @curator = create(:user, role: 'admin', tenant_id: 'dryad')
+    @author = create(:user, tenant_id: 'dryad', role: 'superuser')
+    @document_list = []
+  end
+
+  after(:each) do
+    FileUtils.rm(HOLD_SUBMISSIONS_PATH) if File.exist?(HOLD_SUBMISSIONS_PATH)
+  end
+
+  describe :submitting_quickly do
+
+    before(:each, js: true) do
+      ActionMailer::Base.deliveries = []
+      # Sign in and create a new dataset
+      sign_in(@author)
+      visit root_path
+      click_link 'My Datasets'
+      3.times do
+        start_new_dataset
+        fill_required_fields
+        navigate_to_review
+        check 'agree_to_license'
```

Comment:
Not a big deal, but for future reference you could instead just use the `agree_to_everything` helper method